packages/voting/lib/queries.js

```diff
@@ -0,0 +1,187 @@
+const debug = require('debug')('votings:lib:queries')
+
+const tableMapping = {
+  votings: {
+    name: 'votings',
+    foreignKey: 'votingId',
+    ballotsTable: 'ballots',
+    allowedMembershipsTable: 'votingMembershipRequirements'
+  },
+  elections: {
+    name: 'elections',
+    foreignKey: 'electionId',
+    ballotsTable: 'electionBallots',
+    allowedMembershipsTable: 'electionMembershipRequirements'
+  }
+}
+
+const buildQueries = (tableName) => {
+  const table = tableMapping[tableName]
+  if (!table) {
+    throw new Error(`tableMapping for table name "${table}" not found`)
+  }
+
+  const insertAllowedMemberships = async (id, allowedMemberships, pgdb) => {
+    return Promise.all(
+      allowedMemberships.map(mr =>
+        pgdb.public[table.allowedMembershipsTable].insert({
+          [table.foreignKey]: id,
+          ...mr
+        })
+      )
+    )
+  }
+
+  const findQuery = (where = '') => `
+    SELECT
+      e.*,
+      json_agg(mr.* ORDER BY mr."createdBefore")
+        FILTER (WHERE mr."${table.foreignKey}" IS NOT NULL)
```

Comment:
so that I dont get any nulls in the array.
without: `{ slug: "asdf", allowedMemberships: [null, null], ... }`
with: `{ slug: "asdf", allowedMemberships: [], ... }`

Reply:
nono, `JOIN` only selects rows which have both fiels, `LEFT JOIN` specifically leaves the "left" rows which have no value on the "right" in the results, which is what I want in this case, only that I don't want the null values of the missing "right" values in the array.